### PR TITLE
Raiden --showconfig flag

### LINKED ISSUE #2708: Add cli option `--showconfig`
## Abstract

We should add a cli flag `raiden [OPTIONS...] [ARGS...] --showconfig`. This will dump all configuration values in a standardized format.

## Motivation

Raiden has several configuration items that change its behavior. These may come from different places, i.e.:
- `config.toml`
- `settings.py`
- `constants.py`
- commandline flags

When debugging a client it should be easy to query those values into a standardized format.

Since some values can depend on the client's runtime environment, I propose to use a flag to the main `raiden` command, instead of a `raiden showconfig` subcommand.

## Specification

- Add a cli option `--showconfig` (defaults to `False`)
- Ensure `--showconfig` requests all potentially environment specific flags that may change behavior (i.e. if there are different confirmation block settings for different `--network-id` or `--network-type` values).
- ~~Since we don't have a `--version` flag, this could also include the client, `raiden-contracts` and python version.~~ Also include `raiden version` output.
- Format the output in reasonable subsections, so that it is easy to follow where a specific value comes from/belongs to, i.e.
```
# CONFIG
# SETTINGS
# CONSTANTS
# FLAGS
# VERSIONS
```

## Related
#2707

### PR DESCRIPTION
**Pull request by rakanalh**
Resolves #2708 

Depends on #2715

### CODE REVIEW

**Comment by rakanalh** [comment]
@hackaugusto the idea was that yaml displays better than JSON... and Toml looks more like `.conf` style with section headers which i don't like personally.

**Comment by ulope** [comment]
While I agree that there is not really a need for using yaml, the safety argument doesn't really come into play here since we're only writing yaml.

(And generally `yaml.safe_load()` is the way to avoid the arbitrary code execution problem ;))

**Comment by rakanalh** [comment]
Alright, i'll update it to use Toml instead.

**Comment by LefterisJP** [comment]
@rakanalh needs rebase

**Comment by rakanalh** [comment]
Rebased + Changed config dump format to Toml

File: constraints.txt [hunk 1]
```diff
@@ -65,6 +65,7 @@ pystun-patched-for-raiden==0.1.0
 pytest==3.8.0
 pytoml==0.1.19
 pytz==2018.5
+pyyaml==3.13
```
Comment (hackaugusto) [inline constraints.txt]: we dont need this dependency anymore

File: requirements.txt [hunk 2]
```diff
@@ -22,6 +22,7 @@ py-solc
 pystun-patched-for-raiden
 pytest
 pytoml
+pyyaml
```
Comment (hackaugusto) [inline requirements.txt]: we dont need this dependency anymore

File: raiden/ui/config.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import inspect
+from enum import Enum
+
+import toml
```
Comment (konradkonrad) [inline raiden/ui/config.py]: we don't have `toml` in requirements :( you should use `pytoml`

**Pull request merged**